README.md

```diff
@@ -2,38 +2,285 @@
+2. login with Azure CLI, set the subscription and make sure the `GetCertificate` and `Sign` permission have been granted to your role:
+   ```sh
+   az login
+   az account set --subscription $subscriptionID
+   ```
+3. create an Azure Key Vault and a self-signed certificate:
+   ```sh
+   resourceGroup=notationResource
+   keyVault=notationKV
+   location=westus
+   certName=notationSelfSignedCert
 
+   # create a resource group
+   az group create -n $resourceGroup -l $location
+   
+   # create a Azure Key Vault
+   az keyvault create -l $location -n $keyVault --resource-group $resourceGroup
 
+   # generate certificate policy
+   cat <<EOF > ./selfSignedPolicy.json
+   {
+     "issuerParameters": {
+       "certificateTransparency": null,
+       "name": "Self"
+     },
+     "keyProperties": {
+       "curve": null,
+       "exportable": false,
+       "keySize": 2048,
+       "keyType": "RSA",
+       "reuseKey": true
+     },
+     "secretProperties": {
+       "contentType": "application/x-pem-file"
+     },
+     "x509CertificateProperties": {
+        "ekus": [
+           "1.3.6.1.5.5.7.3.3"
+       ],
+       "keyUsage": [
+         "digitalSignature"
+       ],
+       "subject": "CN=Test Signer",
+       "validityInMonths": 12
+     }
+   }
+   EOF
+
+   # create self-signed certificate
+   az keyvault certificate create -n $certName --vault-name $keyVault -p @selfSignedPolicy.json
+
+   # get the key identifier
+   keyID=$(az keyvault certificate show -n $certName --vault-name $keyVault --query 'kid' -o tsv)
+   ```
+4. run a local registry and push an image to be signed:
+   ```sh
+   # run a local registry with Docker
+   docker run --rm -d -p 5000:5000 ghcr.io/oras-project/registry:v1.0.0-rc.4
+   
+   # push a hello-world image
+   docker pull hello-world
+   docker tag hello-world:latest localhost:5000/hello-world:v1
+   docker push localhost:5000/hello-world:v1
+   ```
+5. notation sign:
+   ```sh
+   # sign with azure-kv
+   notation sign localhost:5000/hello-world:v1 --id $keyID --plugin azure-kv
+   # example output
+   Warning: Always sign the artifact using digest(@sha256:...) rather than a tag(:v1) because tags are mutable and a tag reference can point to a different artifact than the one signed.
+   Successfully signed localhost:5000/hello-world@sha256:f54a58bc1aac5ea1a25d796ae155dc228b3f0e11d046ae276b39c4bf2f13d8c4
+   ```
+6. notation verify:
+   ```sh
+   # add self-signed certificate to notation trust store
+   cat <<EOF > ./selfSignedCert.crt
+   -----BEGIN CERTIFICATE-----
+   $(az keyvault certificate show -n $certName --vault-name $keyVault --query 'cer' -o tsv)
+   -----END CERTIFICATE-----
+   EOF
+   notation cert add --type ca --store selfSigned ./selfSignedCert.crt
+   
+   # add notation trust policy
+   notationConfigDir="${HOME}/.config/notation"                      # for Linux
+   notationConfigDir="${HOME}/Library/Application Support/notation"  # for MacOS
+
```

Comment:
Why Windows is missing here?

Reply:
Because the guide are written in bash script which is supported in Linux and MacOS. If I want to also include Windows Powershell script, many lines will be different for Bash and Powershell.

Reply:
There's a bit of a disconnect here. Earlier in the article, you have PowerShell examples, but then go only Bash. If a user has been running through this using PowerShell, they can't easily proceed. 

You should add PowerShell versions for the steps, or confirm it works in WSL and call that out.

Reply:
Updated to mention that the script supports Windows WSL.